code/go/0chain.net/smartcontract/storagesc/config_settigns.go

```diff
@@ -548,18 +555,57 @@ func (ssc *StorageSmartContract) updateSettings(
 			"unauthorized access - only the owner can update the variables")
 	}
 
+	var newChanges smartcontract.StringMap
+	if err = newChanges.Decode(input); err != nil {
+		return "", common.NewError("update_settings", err.Error())
+	}
+
+	if len(newChanges.Fields) == 0 {
+		return "", nil
+	}
+
+	updateChanges, err := getSettingChanges(balances)
+	if err != nil {
+		return "", common.NewError("update_settings, getting setting changes", err.Error())
+	}
+
+	for key, value := range newChanges.Fields {
+		updateChanges.Fields[key] = value
+	}
+
+	_, err = balances.InsertTrieNode(settingChangesKey, updateChanges)
+	if err != nil {
+		return "", common.NewError("update_settings", err.Error())
+	}
+
+	return "", nil
+}
+
+func (ssc *StorageSmartContract) commitSettingChanges(
+	t *transaction.Transaction,
+	_ []byte,
+	balances chainState.StateContextI,
+) (resp string, err error) {
+	if t.ClientID != balances.GetBlock().MinerID {
```

Comment:
This will stop the transaction being executed by other miners. If a miner recives a block that is generated by others, and the block contains this transaction, the miner would fail to execute it. Therefore, this block could never be verified by others and will not be accepted by the blockchain.

Reply:
I was copying the check from the [payFees transaction](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/minersc/fees.go#L358). Is the situation different here?

Reply:
The payFees transaction will be created after executing all the transactions in a block, and in the payFees transaction, the `ClientID` will be set to `MinerID` as we will only pay fee to the miner who generated the block. So the `if t.ClientID != balances.GetBlock().MinerID` is a must here, but for other normal transactions, we should not do this.